identify_monographs.py

```diff
@@ -0,0 +1,402 @@
+            return False
+        return True
+    # if the record has a production type in FileMaker, check if it's a non-monograph type
+    if _has_non_monograph_fm_production_type(fm_record):
+        return False
+    return True
+
+
+def get_monograph_forms(alma_record: dict, fm_record: dict) -> dict:
+    """For use with monograph records only. Given an Alma record and a FileMaker record,
+    returns a dictionary of monograph forms with data sources (Filemaker or Alma) as keys.
+    """
+    forms = {}
+    if _has_fm_production_type(fm_record):
+        production_types = _get_fm_monograph_production_types(fm_record)
+        normalized_production_types = _normalize_fm_monograph_production_types(
+            production_types
+        )
+        forms["filemaker"] = normalized_production_types
+    monograph_forms = _get_alma_monograph_forms(alma_record)
+    normalized_monograph_forms = _normalize_alma_monograph_form_values(monograph_forms)
+    forms["alma"] = normalized_monograph_forms
+    return forms
+
+
+def compare_monograph_forms(alma_record: dict, fm_record: dict) -> dict:
+    """Compares monograph forms between Alma and FileMaker records."""
+    forms = get_monograph_forms(alma_record, fm_record)
+    alma_forms = forms.get("alma", [])
+    fm_forms = forms.get("filemaker", [])
+    common_forms = set(alma_forms).intersection(set(fm_forms))
+    unique_alma_forms = set(alma_forms).difference(set(fm_forms))
+    unique_fm_forms = set(fm_forms).difference(set(alma_forms))
+    comparison = {
+        "MMS ID": alma_record["001"].data,
+        "common": list(common_forms),
+        "unique_alma": list(unique_alma_forms),
+        "unique_fm": list(unique_fm_forms),
+    }
+    return comparison
+
+
+def _write_form_mismatch_report(monograph_records: list, output_file: str) -> None:
+    """Writes a form mismatch report to a CSV file."""
+    comparison_data = []
+    for alma_record, fm_record in monograph_records:
+        comparison = compare_monograph_forms(alma_record, fm_record)
+        comparison_data.append(comparison)
+    # filter to only records with form mismatches
+    comparison_data = [
+        record
+        for record in comparison_data
+        if record["unique_alma"] or record["unique_fm"]
+    ]
+    print(f"Found {len(comparison_data)} records with form mismatches")
+    with open(output_file, "w") as f:
+        writer = csv.DictWriter(
+            f,
+            fieldnames=["MMS ID", "common", "unique_alma", "unique_fm"],
+        )
+        writer.writeheader()
+        writer.writerows(comparison_data)
+
+
+def _write_monograph_records(records: list, output_file: str) -> None:
+    """Writes monograph records to a MARC file."""
+    with open(output_file, "wb") as f:
+        for alma_record, fm_record in records:
+            f.write(alma_record.as_marc())
+
+
+def main():
+    args = _get_args()
+    alma_data_refs = json.load(open(args.alma_holdings_file))
+    fm_data = csv.DictReader(
```

Comment:
Similar to the alma data refs, I'd move the two commands which get `fm_data` to a helper method. This hides the details of opening file, field names, and list building, none of which needs to be in the main method.